### PR TITLE
Implement LWG-3557: The `static_cast` expression in `convertible_to` has the wrong operand

### LINKED ISSUE #2394: LWG-3557 The `static_cast` expression in `convertible_to` has the wrong operand
LWG-3557 The `static_cast` expression in `convertible_to` has the wrong operand

This was voted into the Working Paper in October 2021, see #2236.

### PR DESCRIPTION
**Pull request by fsb4000**
Fixes #2394

### CODE REVIEW

**Review by miscco — APPROVED** [review]

File: stl/inc/concepts [hunk 1]
```diff
@@ -41,8 +41,8 @@ concept derived_from = __is_base_of(_Base, _Derived)
 
 template <class _From, class _To>
 concept convertible_to = __is_convertible_to(_From, _To)
-    && requires(add_rvalue_reference_t<_From> (&_Fn)()) {
-        static_cast<_To>(_Fn());
+    && requires {
+        static_cast<_To>(declval<_From>());
```
Comment (miscco) [inline stl/inc/concepts:45]: Missing
```suggestion
        static_cast<_To>(_STD declval<_From>());
```

**Review by sam20908 — APPROVED** [review]

**Review by StephanTLavavej — APPROVED** [review]

**Review by CaseyCarter — APPROVED** [review]
For posterity: I'm fine accepting this without test changes because the before and after conditions have equivalent observable behavior. This change is really a specification tweak masquerading as a change in product code.

**Comment by StephanTLavavej** [comment]
I'm mirroring this to the MSVC-internal repo - please notify me if any further changes are pushed.

**Comment by StephanTLavavej** [comment]
Thanks for implementing *another* LWG issue resolution! :rocket: :rocket: :heart_eyes_cat:

**Pull request merged**